lib/network/level-net/index.js

```diff
@@ -0,0 +1,343 @@
+     * @param {Logger} params.logger logger object
+     * @param {Number} [params.streamMaxPendingAck] max number of
+     *   in-flight output stream packets sent to the server without an ack
+     *   received yet
+     * @param {Number} [params.streamAckTimeoutMs] timeout for receiving
+     *   an ack after an output stream packet is sent to the server
+     */
+    constructor(url, params) {
+        assert(params.logger);
+        this.logger = params.logger;
+        this.socket = ioClient(url);
+        this.socketStreams = sioStream.createSocket(
+            this.socket,
+            params.logger,
+            params.streamMaxPendingAck,
+            params.streamAckTimeoutMs);
+        this.socket.on('error', err => {
+            this.logger.warn('connectivity error to storage daemon',
+                             { error: err });
+            return undefined;
+        });
+    }
+
+    /**
+     * @brief internal RPC implementation w/o timeout
+     *
+     * @param {String} remoteCall name of the remote function to call
+     * @param {Array} args list of arguments to the remote function
+     * @param {function} cb callback called when done
+     * @return {undefined}
+     */
+    _call(remoteCall, args, cb) {
+        const wrapCb = (err, data) => {
+            cb(reconstructError(err),
+               this.socketStreams.decodeStreams(data));
+        };
+        this.logger.debug('remote call', { remoteCall, args });
+        this.socket.emit('call', remoteCall,
+                         this.socketStreams.encodeStreams(args), wrapCb);
+        return undefined;
+    }
+
+    /**
+     * @brief call a remote function named @a remoteCall, with
+     * arguments @a args and callback @a cb
+     *
+     * @a cb is called when the remote function returns an ack, or
+     * when the timeout set by @a timeoutMs expires, whichever comes
+     * first. When an ack is received, the callback gets the arguments
+     * sent by the remote function in the ack response. In the case of
+     * timeout, it's passed a single Error argument with the code:
+     * 'ETIMEDOUT' property, and a self-described string in the 'info'
+     * property.
+     *
+     * @param {String} remoteCall name of the remote function to call
+     * @param {Array} args list of arguments to the remote function
+     * @param {function} cb callback called when done or timeout
+     * @param {Number} timeoutMs timeout in milliseconds
+     * @return {undefined}
+     */
+    callTimeout(remoteCall, args, cb, timeoutMs = DEFAULT_CALL_TIMEOUT_MS) {
+        if (typeof cb !== 'function') {
+            throw new Error(`argument cb=${cb} is not a callback`);
+        }
+        async.timeout(this._call.bind(this), timeoutMs,
+                      `operation ${remoteCall} timed out`)(remoteCall,
+                                                           args, cb);
+        return undefined;
+    }
+}
+
+const dbAsyncCommandNames = ['put', 'get', 'del', 'batch'];
```

Comment:
I know it has been merged, but I think even with a small number of keys it's still a good practice here to use an object (O(1)) instead of an array (O(n)) as a dictionary.